src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -343,10 +352,10 @@ const bool Terminal::_isWordDelimiter(std::wstring_view cellChar) const
 // - the corresponding location on the buffer
 const COORD Terminal::_ConvertToBufferCell(const COORD viewportPos) const
 {
-    // don't change the value if at/outside the boundary
-    if (viewportPos.X <= 0 || viewportPos.X >= _buffer->GetSize().RightInclusive())
+    // Invalid position if outside of boundaries
+    if (viewportPos.X < 0 || viewportPos.X > _buffer->GetSize().RightInclusive())
     {
-        return viewportPos;
+        THROW_HR(E_INVALIDARG);

```

Comment:
y new except

Reply:
Theoretically, we should never hit this case. But, if somebody somehow did a double click OUTSIDE of the terminal, I think it makes sense to clamp it to the nearest valid buffer cell. So how does this sound?

```
positionWithOffsets.X = std::clamp(viewportPos.X, static_cast<SHORT>(0), _buffer->GetSize().RightInclusive());
positionWithOffsets.Y = std::clamp(viewportPos.Y, static_cast<SHORT>(0), _buffer->GetSize().BottomInclusive());
```

Note that this is explicitly should not be clamped to the viewport to allow for selections outside of the region we can see. This mainly affects scroll interactions (i.e.: auto scroll, scrolling into an existing selection, etc...).